torch/distributions/kl.py

```diff
@@ -172,14 +172,14 @@ def _kl_beta_beta(p, q):
 @register_kl(Dirichlet, Dirichlet)
 def _kl_dirichlet_dirichlet(p, q):
     # From http://bariskurt.com/kullback-leibler-divergence-between-two-dirichlet-and-beta-distributions/
-    sum_p_alpha = p.alpha.sum(-1)
-    sum_q_alpha = q.alpha.sum(-1)
+    sum_p_alpha = p.alpha.sum(0)
```

Comment:
I'm pretty sure you want `.sum(-1)` here and below. What motivated the change?

Reply:
The tests fails:
```
RuntimeError: inconsistent tensor size, expected r_ [4 x 4 x 3], t [4 x 4 x 3] and src [4 x 4] to have the same number of elements, but got 48, 48 and 16 elements respectively at /media/vishwak/Official/Projects/pytorch/aten/src/TH/generic/THTensorMath.c:1018
```

Reply:
Hmm something is really wrong though. `Dirichlet` even_shape is on the rightmost dimension, as with all other distributions. Maybe I initialized Dirichlet incorrectly? Does this fail for Beta or Dirichlet tests?

Reply:
It fails for `Dirichlet` alone.